w11h03/test/pgdp/networking/SocketTest.java

```diff
@@ -0,0 +1,455 @@
+    private HTTPClientMock httpClient;
+    private Throwable lastThrowable = null;
+
+    // Messages
+    private static final byte[] SERVER_HELLO = new byte[]{0x00, 0x00, 0x2a};
+    private static final byte[] AUTHENTICATION_FAILURE = new byte[]{0x00, (byte) 0xf1};
+    private static final byte[] SERVER_ACK = new byte[]{0x00, 0x05};
+    private static final byte[] SERVER_TIMEOUT = new byte[] {(byte) 0x00, (byte) 0xff};
+
+    @BeforeAll
+    static void setupAll() {
+        try {
+            server = new ServerSocket(1337);
+        } catch (IOException e) {
+            throw new RuntimeException(
+                    "Something went wrong while creating the local SocketServer. Got anything running on port 1337?\n"
+                            + "Error:\n" + e
+            );
+        }
+        System.out.println("""
+                If you are using a slower device and your SocketTests keep failing for no reason, try increasing the SOCKET_TIMEOUT.""");
+    }
+
+    @BeforeEach
+    void setup() {
+        this.httpClient = new HTTPClientMock();
+        httpClient.respond(200, """
+                {
+                  "access_token": "exampleToken",
+                  "token_type": "irrelevant"
+                }""");
+        setField(dataHandler, "client", httpClient);
+        this.connect = getMethod(dataHandler, "connect");
+        // Set socket to localhost
+        setField(dataHandler, "serverAddress", "127.0.0.1");
+        setField(dataHandler, "username", "Alfred");
+        setField(dataHandler, "password", "Wagner");
+        setField(dataHandler, "id", 84342);
+    }
+
+    @Test
+    @DisplayName("connect() – successful")
+    void testConnectSuccess() throws IOException {
+        warnConnect();
+
+        // Execute method and read socket output
+        Thread clientThread = new Thread(this::connect);
+        clientThread.start();
+        Socket client = server.accept();
+        client.getOutputStream().write(SERVER_HELLO);
+        client.getOutputStream().flush();
+        var buffer = getAllBytes(clientThread, client);
+
+        // Assert no Exception
+        assertNull(lastThrowable, "connect() threw an exception when it wasn't supposed to.");
+
+        // Assert ClientHello
+        final byte[] expectedHello = new byte[] {0x00, 0x01};
+        byte[] actualHello = copyOf(buffer, 0, 2, "ClientHello");
+        assertArrayEquals(expectedHello, actualHello, "Did not receive ClientHello as expected.");
+
+        // Assert ClientIdentification
+        byte[] clientIdentification = new byte[] {0x00, 0x02};
+        byte[] actualIdentification = copyOf(buffer, 2, 4, "ClientIdentification");
+        assertArrayEquals(clientIdentification, actualIdentification, "Did not receive ClientIdentification as expected.");
+        byte idSize = copyOf(buffer, 4, 5, "ClientIdentification k")[0];
+        int id = bufferToInt(copyOf(buffer, 5, 5 + idSize, "ClientIdentification id"));
```

Comment:
Hier eine Hardgecodete id und entsprechend die erwartete antwort. Ich weiß ist nicht schön aber es soll ja nicht allzu viel verraten werden

Reply:
Achso okay. Also bufferToInt und so entfernen und einfach alle Möglichkeiten durchprobieren?

Reply:
Genau, ich weiß is nicht schön.

Reply:
Ist es okay, wenn ich das mit der `idSize` beibehalte und so dynamisch den `buffer` auslese, weil sonst weiß ich nicht welche slice vom Buffer die `ClientIdentification` ist.

Reply:
Wenn du eine zuverlässige Antwort haben möchtest, müsstest du Konrad auf Zulip per DM fragen

Reply:
ja klar, das ist ja nur ein byte

Reply:
Asoo hahaha, wusste nicht, dass du Konrad bist xD